### DISCUSSION TITLE
play mp3 in SPIFFS of ESP32

### DISCUSSION THREAD

**Discussion by liuquanli1970** [schreibfaul1/ESP32-audioI2S#147] [Q&A]
Audioi2s is the easiest library to use at present, which is very popular with beginners. Thank you, Wolle.
If I upload MP3 files to SPIFFS of esp32 instead of SD, what do I need to do to play MP3 in SPIFFS?
 I already have I2S amplifier.

Best regards,
Thank you!

**Comment by schreibfaul1** [comment]
That works, some preparations have to be made beforehand:
- Choose a partition scheme that is large enough and supports SPIFFS
![image](https://user-images.githubusercontent.com/26044260/114444926-b6107700-9bcf-11eb-96c9-fe1f202ffc8b.png)
- in *.ino  ```#include "SPIFFS.h"```
- in setup()  ```SPIFFS.begin();``` and ``` audio.connecttoFS(SPIFFS, "test.mp3");```
- and load 'test.mp3' into SPIFFS beforehand
